tools/release/adot-operator-images-mirror/main.go

```diff
@@ -71,27 +67,25 @@ func main() {
 	}
 
 	log.Println("Creating Docker client")
-	var client DockerClient
-	client, err = createDockerClient()
+	dc, err := docker.NewClientWithOpts(docker.FromEnv, docker.WithAPIVersionNegotiation())
 	if err != nil {
 		log.Fatalf("Could not create Docker client: %v", err)
 	}
 
-	info, err := client.Info()
+	info, err := dc.Info(ctx)
 	if err != nil {
 		log.Fatalf("Could not get Docker info: %v", err)
 	}
 	log.Printf("Connected to Docker daemon: %s @ %s", info.Name, info.ServerVersion)
 
 	log.Println("Creating AWS client")
-	cfg, err := awsconfig.LoadDefaultConfig(context.TODO())
+	// Override the AWS region with the ecrPublicRegion for ECR authentication.
+	cfg, err := awsconfig.LoadDefaultConfig(ctx, awsconfig.WithRegion(ecrPublicRegion))
```

Comment:
I like that you are add thing region on struct creation instead later in the code